File: Source/Examples/Beispielanwendung_VB/customDownloadDialog.vb
Public Class customDownloadDialog

    Public Sub New()

        ' Dieser Aufruf ist für den Windows Form-Designer erforderlich.
        InitializeComponent()

        ' Fügen Sie Initialisierungen nach dem InitializeComponent()-Aufruf hinzu.
        Font = SystemFonts.MessageBoxFont

    End Sub

    Private Sub customDownloadDialog_downloadUpdatesCompleted(ByVal sender As Object, ByVal e As System.ComponentModel.AsyncCompletedEventArgs) Handles Me.downloadUpdatesCompleted

        If Not e.Error Is Nothing Then
            If Not e.Cancelled Then Throw e.Error
            DialogResult = Windows.Forms.DialogResult.Cancel
        Else
            DialogResult = Windows.Forms.DialogResult.OK
        End If
        Close()

    End Sub

    Private Sub customDownloadDialog_downloadUpdatesProgressChanged(ByVal sender As Object, ByVal e As updateSystemDotNet.appEventArgs.downloadUpdatesProgressChangedEventArgs) Handles Me.downloadUpdatesProgressChanged
        prgDownload.Value = e.ProgressPercentage
    End Sub

    Private Sub customDownloadDialog_Shown(ByVal sender As Object, ByVal e As System.EventArgs) Handles Me.Shown
        startDownload()
    End Sub

    Private Sub btnCancelUpdate_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCancelUpdate.Click
        If isBusy Then cancelDownload()
    End Sub
End Class

File: Source/Examples/Beispielanwendung_CSharp/customDownloadDialog.cs
using System;
using System.ComponentModel;
using System.Drawing;
using System.Windows.Forms;
using updateSystemDotNet;
using updateSystemDotNet.appEventArgs;

namespace Beispielanwendung_CSharp {
	public sealed partial class customDownloadDialog : updateDownloadBaseForm {
		public customDownloadDialog() {
			InitializeComponent();
			Font = SystemFonts.MessageBoxFont;

			downloadUpdatesCompleted += customDownloadDialog_downloadUpdatesCompleted;
			downloadUpdatesProgressChanged += customDownloadDialog_downloadUpdatesProgressChanged;
			Shown += customDownloadDialog_Shown;
		}

		private void customDownloadDialog_Shown(object sender, EventArgs e) {
			startDownload();
		}

		private void customDownloadDialog_downloadUpdatesProgressChanged(object sender,
		                                                                 downloadUpdatesProgressChangedEventArgs e) {
			prgDownload.Value = e.ProgressPercentage;
		}

		private void customDownloadDialog_downloadUpdatesCompleted(object sender, AsyncCompletedEventArgs e) {
			if (e.Error != null) {
				if (!e.Cancelled)
					MessageBox.Show(string.Format("Whoops, da ist was beim Download schiefgegangen:\r\n{0}", e.Error.Message),
					                "Test Anwendung",
					                MessageBoxButtons.OK, MessageBoxIcon.Error);

				DialogResult = DialogResult.Cancel;
			}
			else
				DialogResult = DialogResult.OK;

			Close();
		}

		private void btnCancelUpdate_Click(object sender, EventArgs e) {
			if (isBusy)
				cancelDownload();
		}
	}
}